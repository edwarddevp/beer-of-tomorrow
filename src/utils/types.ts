export interface RequestType<T> {
  isLoading: boolean;
  data: T;
  errors: Array<{ message: string }>;
}

export interface RequestListType<T> {
  isLoading: boolean;
  data: T;
  errors: Array<{ message: string }>;
  filter: string;
  options: {
    page: number;
    per_page: number;
  };
  noMoreItems: boolean;
}

export interface Data {
  value: string;
  label: string;
}
export interface Beer {
  id: number;
  name: string;
  tagline: string;
  first_brewed: string;
  description: string;
  image_url: string;
  abv: number;
  ibu: number;
  target_fg: number;
  target_og: number;
  ebc: number;
  srm: number;
  ph: number;
  attenuation_level: number;
  volume: {
    value: number;
    unit: string;
  };
  boil_volume: {
    value: number;
    unit: string;
  };
  method: {
    mash_temp: {
      temp: {
        value: number;
        unit: string;
      };
      duration: number | null;
    }[];
    fermentation: {
      temp: {
        value: number;
        unit: string;
      };
    };
    twist: null;
  };
  ingredients: {
    malt: {
      name: string;
      amount: {
        value: number;
        unit: string;
      };
    }[];
    hops: {
      name: string;
      amount: {
        value: number;
        unit: string;
      };
      add: string;
      attribute: string;
    }[];
    yeast: string;
  };
  food_pairing: string[];
  brewers_tips: string;
  contributed_by: string;
}

export interface AnyObject {
  [prop: string]: any;
}
