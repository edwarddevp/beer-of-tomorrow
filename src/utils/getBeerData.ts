import { CartBeer } from "src/reducers/cart";

export const getBeerData = (beer: Partial<CartBeer>) => {
  const isInOffert = beer.ibu ? beer.ibu > 40 : false;
  const beerNameSplit = beer.name ? beer.name.split(" ") : "";
  const beerName =
    beerNameSplit.length > 1
      ? beerNameSplit?.[0] + " " + beerNameSplit?.[1]
      : beer.name;
  return { isInOffert, beerName };
};
