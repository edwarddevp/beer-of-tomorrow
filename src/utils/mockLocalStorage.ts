/* istanbul ignore file */
// @ts-ignore
class LocalStorageMock {
  store: any;
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key: string) {
    return this.store[key] || null;
  }

  setItem(key: string, value: any) {
    this.store[key] = String(value);
  }

  removeItem(key: string) {
    delete this.store[key];
  }
}
// @ts-ignore
global.localStorage = new LocalStorageMock();
