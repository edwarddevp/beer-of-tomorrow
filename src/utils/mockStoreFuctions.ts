import { AppDispatch, RootState } from "./../configs/store";

export const create = () => {
  const store = {
    getState: jest.fn(() => ({})),
    dispatch: jest.fn((x) => x),
  };
  const next = jest.fn();

  const invoke = async (action: Function) => await thunk(store)(next)(action);

  return { store, next, invoke };
};

interface MockThunk {
  dispatch: AppDispatch;
  getState: RootState;
}

export const thunk =
  ({ dispatch, getState }: MockThunk) =>
  (next: Function) =>
  async (action: Function) => {
    if (typeof action === "function") {
      return await action(dispatch, getState);
    }

    return await next(action);
  };
