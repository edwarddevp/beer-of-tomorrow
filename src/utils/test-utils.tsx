import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "../theme";
import { createSerializer, matchers } from "@emotion/jest";
import { useStore } from "@/configs/store";
import { Provider } from "react-redux";
import mockBeerItems from "./mock-beer-items";
// @ts-ignore
expect.addSnapshotSerializer(createSerializer());
// @ts-ignore
expect.extend(matchers);

const initialStoreValues = {
  getBeerById: {
    data: mockBeerItems[0],
    isLoading: false,
    errors: [],
  },
};

const AllTheProviders: FC = ({ children }) => {
  const store = useStore(initialStoreValues);
  return (
    <ChakraProvider theme={theme}>
      <Provider store={store}>{children}</Provider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
