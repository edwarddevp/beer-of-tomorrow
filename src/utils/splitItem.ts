import { AnyObject } from "@/utils/types";

// helper to get objects subkeys by a string
export const splitItem = (label: string, row: AnyObject) => {
  label = label.replace(/\[(\w+)]/g, ".$1"); // convert indexes to properties
  label = label.replace(/^\./, ""); // strip a leading dot
  const a = label.split(".");
  for (let i = 0, n = a.length; i < n; ++i) {
    const k = a[i];
    if (k in row) {
      row = row[k];
    } else {
      return;
    }
  }
  return row;
};
