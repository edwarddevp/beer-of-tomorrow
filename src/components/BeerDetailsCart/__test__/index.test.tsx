import React from "react";
// Using render and screen from test-utils.js instead of
// @testing-library/react
import { fireEvent, prettyDOM, render, within } from "test-utils";
import { BeerDetailsCart } from "../";
import "@/utils/mockLocalStorage";

const sleep = (milliseconds: number) => {
  return new Promise((resolve) => setTimeout(resolve, milliseconds));
};
describe("BeerDetailsCart", () => {
  it("renders", () => {
    const { container } = render(<BeerDetailsCart />);
    expect(container).toMatchSnapshot();
  });
  it("handleAdd", async () => {
    const { getByLabelText } = render(<BeerDetailsCart noToast />);
    const counter = within(getByLabelText("Item Quantity in the Cart"));
    counter.getByText("0");
    const addButton = getByLabelText("Add");
    fireEvent.click(addButton);
    await sleep(500);
    counter.getByText("1");
  });
  it("handleSubstrack", async () => {
    const { getByLabelText } = render(<BeerDetailsCart noToast />);
    const counter = within(getByLabelText("Item Quantity in the Cart"));
    counter.getByText("1");
    const addButton = getByLabelText("Add");
    const Substrack = getByLabelText("Add");
    fireEvent.click(addButton);
    fireEvent.click(addButton);
    fireEvent.click(Substrack);
    await sleep(500);
    counter.getByText("2");
  });
});
