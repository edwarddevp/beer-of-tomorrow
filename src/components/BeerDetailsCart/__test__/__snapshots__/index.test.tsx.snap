// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDetailsCart renders 1`] = `
.emotion-0 {
  display: grid;
  grid-gap: var(--chakra-space-4);
  grid-template-columns: 1fr;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  background: var(--chakra-colors-backgrounds-cards);
  border-width: 2px;
  border-color: var(--chakra-colors-backgrounds-cardsBorders);
  -webkit-padding-start: var(--chakra-space-6);
  padding-inline-start: var(--chakra-space-6);
  -webkit-padding-end: var(--chakra-space-6);
  padding-inline-end: var(--chakra-space-6);
  padding-top: var(--chakra-space-8);
  padding-bottom: var(--chakra-space-8);
  margin-top: var(--chakra-space-8);
  margin-bottom: var(--chakra-space-8);
  box-shadow: var(--chakra-shadows-secondary);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
  max-height: none;
  max-width: none;
  position: relative;
}

@media screen and (min-width: 48em) {
  .emotion-0 {
    grid-template-columns: 1fr 1fr auto;
  }
}

@media screen and (min-width: 62em) {
  .emotion-0 {
    max-height: 120px;
    max-width: 100%;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-3 {
  color: var(--chakra-colors-primary-500);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-4xl);
}

.emotion-4 {
  font-weight: var(--chakra-fontWeights-bold);
  color: var(--chakra-colors-gray-300);
  font-size: var(--chakra-fontSizes-lg);
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
  padding-top: 0px;
}

@media screen and (min-width: 30em) {
  .emotion-4 {
    font-size: var(--chakra-fontSizes-sm);
  }
}

@media screen and (min-width: 48em) {
  .emotion-4 {
    font-size: var(--chakra-fontSizes-xl);
  }
}

.emotion-5 {
  display: grid;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  box-shadow: var(--chakra-shadows-secondary);
  border-radius: 50px;
  min-width: 190px;
  padding-left: var(--chakra-space-6);
}

@media screen and (min-width: 30em) {
  .emotion-6 {
    min-width: 250px;
  }
}

@media screen and (min-width: 48em) {
  .emotion-6 {
    min-width: 190px;
  }
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: auto;
  line-height: 1.2;
  border-radius: 100%;
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: var(--chakra-sizes-6);
  min-width: var(--chakra-sizes-6);
  font-size: var(--chakra-fontSizes-xs);
  -webkit-padding-start: var(--chakra-space-2);
  padding-inline-start: var(--chakra-space-2);
  -webkit-padding-end: var(--chakra-space-2);
  padding-inline-end: var(--chakra-space-2);
  opacity: 1;
  border-width: 1px;
  border-color: var(--chakra-colors-primary-200);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  color: var(--chakra-colors-primary-200);
  padding: 0px;
}

.emotion-7:focus,
.emotion-7[data-focus] {
  box-shadow: var(--chakra-shadows-none);
}

.emotion-7[disabled],
.emotion-7[aria-disabled=true],
.emotion-7[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-7:hover,
.emotion-7[data-hover] {
  background-image: linear-gradient(to top, var(--chakra-colors-primary-200), var(--chakra-colors-primary-100), var(--chakra-colors-primary-50));
  color: var(--chakra-colors-white);
}

.emotion-7:hover[disabled],
.emotion-7[data-hover][disabled],
.emotion-7:hover[aria-disabled=true],
.emotion-7[data-hover][aria-disabled=true],
.emotion-7:hover[data-disabled],
.emotion-7[data-hover][data-disabled] {
  background: linear(to-t, primary.200, primary.100, primary.50);
}

.emotion-7:active,
.emotion-7[data-active] {
  box-shadow: var(--chakra-shadows-none);
  background-image: linear-gradient(to top, var(--chakra-colors-primary-200), var(--chakra-colors-primary-100), var(--chakra-colors-primary-50));
  color: var(--chakra-colors-white);
}

.emotion-8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  justify-self: stretch;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.emotion-11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: 50px;
  line-height: 1.2;
  border-radius: 50px;
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: 50px;
  min-width: var(--chakra-sizes-10);
  font-size: 1.5rem;
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  opacity: 1;
  background-image: linear-gradient(to top, var(--chakra-colors-primary-200), var(--chakra-colors-primary-100), var(--chakra-colors-primary-50));
  color: var(--chakra-colors-white);
  padding: 0px;
  margin-left: var(--chakra-space-4);
}

.emotion-11:focus,
.emotion-11[data-focus] {
  box-shadow: var(--chakra-shadows-none);
}

.emotion-11[disabled],
.emotion-11[aria-disabled=true],
.emotion-11[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-11:hover,
.emotion-11[data-hover] {
  background-image: linear-gradient(to top, var(--chakra-colors-primary-300), var(--chakra-colors-primary-200), var(--chakra-colors-primary-100));
}

.emotion-11:hover[disabled],
.emotion-11[data-hover][disabled],
.emotion-11:hover[aria-disabled=true],
.emotion-11[data-hover][aria-disabled=true],
.emotion-11:hover[data-disabled],
.emotion-11[data-hover][data-disabled] {
  background: linear(to-t, primary.300, primary.200, primary.100);
}

.emotion-11:active,
.emotion-11[data-active] {
  box-shadow: var(--chakra-shadows-none);
  background-image: linear-gradient(to top, var(--chakra-colors-primary-300), var(--chakra-colors-primary-200), var(--chakra-colors-primary-100));
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: absolute;
  white-space: nowrap;
  vertical-align: middle;
  outline: 2px solid transparent;
  outline-offset: 2px;
  width: 18px;
  line-height: 1.2;
  border-radius: var(--chakra-radii-md);
  font-weight: var(--chakra-fontWeights-semibold);
  transition-property: var(--chakra-transition-property-common);
  transition-duration: var(--chakra-transition-duration-normal);
  height: 18px;
  min-width: var(--chakra-sizes-10);
  font-size: 3rem;
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  color: inherit;
  padding: 0px;
  margin-left: var(--chakra-space-4);
  top: var(--chakra-space-5);
  right: 5px;
}

@media screen and (min-width: 80em) {
  .emotion-12 {
    position: relative;
    top: 0px;
    right: 0px;
  }
}

.emotion-12:focus,
.emotion-12[data-focus] {
  box-shadow: var(--chakra-shadows-none);
}

.emotion-12[disabled],
.emotion-12[aria-disabled=true],
.emotion-12[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: var(--chakra-shadows-none);
}

.emotion-12:hover,
.emotion-12[data-hover] {
  background: var(--chakra-colors-gray-100);
}

.emotion-12:hover[disabled],
.emotion-12[data-hover][disabled],
.emotion-12:hover[aria-disabled=true],
.emotion-12[data-hover][aria-disabled=true],
.emotion-12:hover[data-disabled],
.emotion-12[data-hover][data-disabled] {
  background: initial;
}

.emotion-12:active,
.emotion-12[data-active] {
  box-shadow: var(--chakra-shadows-none);
  background: var(--chakra-colors-gray-200);
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="chakra-stack emotion-1"
    >
      <div
        class="chakra-stack emotion-2"
      >
        <p
          class="chakra-text emotion-3"
        >
          $
          1044
        </p>
        <p
          class="chakra-text emotion-4"
        >
          $
          1012.68
        </p>
      </div>
    </div>
    <div
      class="emotion-5"
    >
      <div
        class="emotion-6"
      >
        <button
          aria-label="Substrack"
          class="chakra-button emotion-7"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
        </button>
        <div
          class="emotion-8"
        />
        <span
          aria-label="Item Quantity in the Cart"
        >
          0
        </span>
        <div
          class="emotion-8"
        />
        <button
          aria-label="Add"
          class="chakra-button emotion-7"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
        </button>
        <button
          aria-label="Add To Cart"
          class="chakra-button emotion-11"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 176a16 16 0 00-16 16v216c0 30.24 25.76 56 56 56h272c30.24 0 56-24.51 56-54.75V192a16 16 0 00-16-16zm80 0v-32a96 96 0 0196-96h0a96 96 0 0196 96v32"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
            <path
              d="M160 224v16a96 96 0 0096 96h0a96 96 0 0096-96v-16"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="32"
            />
          </svg>
        </button>
      </div>
    </div>
    <button
      aria-label="Share"
      class="chakra-button emotion-12"
      type="button"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="17.5"
          cy="18.5"
          fill="none"
          r="1.5"
        />
        <circle
          cx="5.5"
          cy="11.5"
          fill="none"
          r="1.5"
        />
        <circle
          cx="17.5"
          cy="5.5"
          fill="none"
          r="1.5"
        />
        <path
          d="M5.5,15c0.91,0,1.733-0.358,2.357-0.93l6.26,3.577C14.048,17.922,14,18.204,14,18.5c0,1.93,1.57,3.5,3.5,3.5 s3.5-1.57,3.5-3.5S19.43,15,17.5,15c-0.91,0-1.733,0.358-2.357,0.93l-6.26-3.577c0.063-0.247,0.103-0.502,0.108-0.768l6.151-3.515 C15.767,8.642,16.59,9,17.5,9C19.43,9,21,7.43,21,5.5S19.43,2,17.5,2S14,3.57,14,5.5c0,0.296,0.048,0.578,0.117,0.853L8.433,9.602 C7.808,8.64,6.729,8,5.5,8C3.57,8,2,9.57,2,11.5S3.57,15,5.5,15z M17.5,17c0.827,0,1.5,0.673,1.5,1.5S18.327,20,17.5,20 S16,19.327,16,18.5S16.673,17,17.5,17z M17.5,4C18.327,4,19,4.673,19,5.5S18.327,7,17.5,7S16,6.327,16,5.5S16.673,4,17.5,4z M5.5,10C6.327,10,7,10.673,7,11.5S6.327,13,5.5,13S4,12.327,4,11.5S4.673,10,5.5,10z"
        />
      </svg>
    </button>
  </div>
</div>
`;
